### ISSUE TITLE
Resnet-50 error related to matrix sizes

### ISSUE THREAD

**Issue by benson31** [LLNL/lbann#769]
I don't know if this is related to #768; I doubt it, but maybe. Resnet-50 on Lassen dies with the following error:
```****************************************************************
LBANN error on rank <each of them> (../../../src/layers/layer.cpp:1017): layer "labels" expected an input tensor stored in a 1000 x 256 matrix from layer "data", but got a 0 x 256 matrix
Stack trace:
   0: lbann::stack_trace::get()
   1: lbann::exception::exception(std::string, bool)
   2: lbann::Layer::fp_setup_inputs(long long)
   3: lbann::Layer::forward_prop()
   4: lbann::model::forward_prop(execution_mode)
   5: lbann::model::train_mini_batch()
   6: lbann::model::train(int, int)
   7: install/bin/lbann() [...] (could not find stack frame symbol)
   8: /lib64/libc.so.6(+...) [...] (could not find stack frame symbol)
   9: __libc_start_main (demangling failed)
****************************************************************```

**Comment by timmoon10** [comment]
I don't reproduce this on Pascal.

**Comment by ndryden** [comment]
Fixed, most likely by #771.
